src/Sylius/Bundle/CoreBundle/Controller/ProductController.php

```diff
@@ -114,6 +115,36 @@ public function indexByTaxonIdAction(Request $request, $id)
         return $this->renderResults($taxon, $paginator, 'productIndex.html', $request->get('page', 1));
     }
 
+    /**
+     * List products categorized under given archetype (fetch by its code or id).
+     *
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws NotFoundHttpException
+     */
+    public function indexByArchetypeAction(Request $request)
+    {
+        $archetypeIdOrCode = $request->attributes->get('archetype');
+        if (filter_var($archetypeIdOrCode, FILTER_VALIDATE_INT)) {
+            $archetype = $this->get('sylius.repository.product_archetype')->find($archetypeIdOrCode);
+        } else {
+            $archetype = $this->get('sylius.repository.product_archetype')->findOneByCode($archetypeIdOrCode);
+        }
+
+        if (null === $archetype) {
+            throw new NotFoundHttpException('Requested archetype does not exist.');
+        }
+
+        $paginator = $this
+            ->getRepository()
```

Comment:
I think it's better way not to call ``$this->getRepository()`` method, but ``$this->get('sylius.repository.product')`` instead.